### ISSUE TITLE
using the openstack driver results in 'No valid host was found.' on docker-machine 0.3.0-rc1

### ISSUE THREAD

**Issue by cglewis** [docker/machine#1261]
The instance errors out and just says `No valid host was found.`

`docker-machine` just sits at 'Creating machine...' even though the instance has been created and errors out on openstack.

**Comment by ggiamarchi** [comment]
`No valid host was found` is not an error from docker machine, it's an OpenStack error. Nova scheduler usually say that when it did not manage to elect a compute node to spawn the instance. To get more information on the reason you should ask to your OpenStack administrator or have a look at the nova scheduler log if you have access to it.

**Comment by cglewis** [comment]
Figured this out.  FWIW, I'll add what I found here in case others run into this:

The first issue was I had one node in my cluster that had issues and therefore had no VMs, and was the highest weighting to get the next instance spun up on it.  Using Horizon, it fails over and picks the next available compute node, however launching from `docker-machine` just fails.

Once I removed that node from the cluster I still had the same problem, but the cause was I was using the UUID of my custom flavor, when it was actually expecting the numerical ID from the original flavors.

It would be really nice if we could use names for both flavors and images in addition to being able to specify the ID.
